### ISSUE TITLE
cascading worldgen lag

### ISSUE THREAD

**Issue by C0rn3j** [Ellpeck/ActuallyAdditions#953]
```
[15:29:30] [Server thread/WARN] [FML]: Recurrent Complex loaded a new chunk (37, -1  Dimension: -52) during chunk population, causing cascading worldgen lag. Please report this to the mod's issue tracker. This log can be disabled in the Forge config.
[15:29:30] [Server thread/WARN] [FML]: Recurrent Complex loaded a new chunk (37, -2  Dimension: -52) during chunk population, causing cascading worldgen lag. Please report this to the mod's issue tracker. This log can be disabled in the Forge config.
[15:30:14] [Server thread/INFO] [powerinventory]: Player File: _f07bf2e9-3dba-30e4-86a0-628a8eae61ad.opi
[15:30:14] [Server thread/INFO] [powerinventory]: Player File: _f07bf2e9-3dba-30e4-86a0-628a8eae61ad.opibk
[15:30:50] [Server thread/WARN] [FML]: Recurrent Complex loaded a new chunk (36, -19  Dimension: -52) during chunk population, causing cascading worldgen lag. Please report this to the mod's issue tracker. This
log can be disabled in the Forge config.

**[15:30:50] [Server thread/WARN] [FML]: Actually Additions loaded a new chunk (38, -18  Dimension: -52) during chunk population, causing cascading worldgen lag. Please report this to the mod's issue tracker. This log can be disabled in the Forge config.**

[15:30:50] [Server thread/WARN] [FML]: Recurrent Complex loaded a new chunk (40, -19  Dimension: -52) during chunk population, causing cascading worldgen lag. Please report this to the mod's issue tracker. This
log can be disabled in the Forge config.
[15:30:59] [Server thread/WARN]: Can't keep up! Did the system time change, or is the server overloaded? Running 9509ms behind, skipping 190 tick(s)
```

Forge 2515. MC 1.12.2, 
ActuallyAdditions-1.12.1-r121.jar
RecurrentComplex-1.4.5.1.jar

+a bunch more mods that dont seem to be causing any issue.

Issue for Recc complex is here - https://github.com/Ivorforce/RecurrentComplex/issues/155

**Comment by Ellpeck** [comment]
Hahahahahahah have fun with this, @Shadows-of-Fire

**Comment by Shadows-of-Fire** [comment]
I can't even fix it until i can figure out what structure it is, so uh, *meh*.
Maybe later, but one chunk getting loaded like that won't set the universe on fire.

**Comment by TwilyS** [comment]
---- Minecraft Crash Report ----

WARNING: coremods are present:
  Inventory Tweaks Coremod (InventoryTweaks-1.63.jar)
  AstralCore (astralsorcery-1.12.2-1.7.2.jar)
Contact their authors BEFORE contacting forge

// Quite honestly, I wouldn't worry myself about that.

Time: 11/3/17 7:44 PM
Description: Unexpected error

java.lang.IllegalStateException: Actually Additions: Something is very wrong.  This method was provided with 
0: String <- CLASS | INSTANCE -> Null ItemStack detected, 
1: null
The current side is: CLIENT
	at de.ellpeck.actuallyadditions.mod.util.AwfulUtil.callTheFuckinPolice(AwfulUtil.java:107)
	at de.ellpeck.actuallyadditions.mod.util.StackUtil.isValid(StackUtil.java:39)
	at de.ellpeck.actuallyadditions.mod.booklet.button.EntryButton.<init>(EntryButton.java:33)
	at de.ellpeck.actuallyadditions.mod.booklet.gui.GuiMainPage.func_73866_w_(GuiMainPage.java:226)
	at net.minecraft.client.gui.GuiScreen.func_146280_a(GuiScreen.java:478)
	at net.minecraft.client.Minecraft.func_147108_a(Minecraft.java:1016)
	at net.minecraftforge.fml.client.FMLClientHandler.showGuiScreen(FMLClientHandler.java:534)
	at net.minecraftforge.fml.common.FMLCommonHandler.showGuiScreen(FMLCommonHandler.java:320)
	at net.minecraftforge.fml.common.network.internal.FMLNetworkHandler.openGui(FMLNetworkHandler.java:111)
	at net.minecraft.entity.player.EntityPlayer.openGui(EntityPlayer.java:2549)
	at de.ellpeck.actuallyadditions.mod.items.ItemBooklet.func_77659_a(ItemBooklet.java:78)
	at net.minecraft.item.ItemStack.func_77957_a(ItemStack.java:217)
	at net.minecraft.client.multiplayer.PlayerControllerMP.func_187101_a(PlayerControllerMP.java:511)
	at PlayerControllerOF.func_187101_a(PlayerControllerOF.java:80)
	at net.minecraft.client.Minecraft.func_147121_ag(Minecraft.java:1595)
	at net.minecraft.client.Minecraft.func_184117_aA(Minecraft.java:2252)
	at net.minecraft.client.Minecraft.func_184118_az(Minecraft.java:2018)
	at net.minecraft.client.Minecraft.func_71407_l(Minecraft.java:1806)
	at net.minecraft.client.Minecraft.func_71411_J(Minecraft.java:1096)
	at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:397)
	at net.minecraft.client.main.Main.main(SourceFile:123)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at net.minecraft.launchwrapper.Launch.launch(Launch.java:135)
	at net.minecraft.launchwrapper.Launch.main(Launch.java:28)


A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Head --
Thread: Client thread
Stacktrace:
	at de.ellpeck.actuallyadditions.mod.util.AwfulUtil.callTheFuckinPolice(AwfulUtil.java:107)
	at de.ellpeck.actuallyadditions.mod.util.StackUtil.isValid(StackUtil.java:39)
	at de.ellpeck.actuallyadditions.mod.booklet.button.EntryButton.<init>(EntryButton.java:33)
	at de.ellpeck.actuallyadditions.mod.booklet.gui.GuiMainPage.func_73866_w_(GuiMainPage.java:226)
	at net.minecraft.client.gui.GuiScreen.func_146280_a(GuiScreen.java:478)
	at net.minecraft.client.Minecraft.func_147108_a(Minecraft.java:1016)
	at net.minecraftforge.fml.client.FMLClientHandler.showGuiScreen(FMLClientHandler.java:534)
	at net.minecraftforge.fml.common.FMLCommonHandler.showGuiScreen(FMLCommonHandler.java:320)
	at net.minecraftforge.fml.common.network.internal.FMLNetworkHandler.openGui(FMLNetworkHandler.java:111)
	at net.minecraft.entity.player.EntityPlayer.openGui(EntityPlayer.java:2549)
	at de.ellpeck.actuallyadditions.mod.items.ItemBooklet.func_77659_a(ItemBooklet.java:78)
	at net.minecraft.item.ItemStack.func_77957_a(ItemStack.java:217)
	at net.minecraft.client.multiplayer.PlayerControllerMP.func_187101_a(PlayerControllerMP.java:511)
	at PlayerControllerOF.func_187101_a(PlayerControllerOF.java:80)
	at net.minecraft.client.Minecraft.func_147121_ag(Minecraft.java:1595)
	at net.minecraft.client.Minecraft.func_184117_aA(Minecraft.java:2252)
	at net.minecraft.client.Minecraft.func_184118_az(Minecraft.java:2018)

-- Affected level --
Details:
	Level name: MpServer
	All players: 1 total; [EntityPlayerSP['Mu_guang_138'/22, l='MpServer', x=-560.88, y=4.82, z=250.87]]
	Chunk stats: MultiplayerChunkCache: 1038, 1038
	Level seed: 0
	Level generator: ID 01 - flat, ver 0. Features enabled: false
	Level generator options: 
	Level spawn location: World: (-452,4,464), Chunk: (at 12,0,0 in -29,29; contains blocks -464,0,464 to -449,255,479), Region: (-1,0; contains chunks -32,0 to -1,31, blocks -512,0,0 to -1,255,511)
	Level time: 137157 game time, 137157 day time
	Level dimension: 0
	Level storage version: 0x00000 - Unknown?
	Level weather: Rain time: 0 (now: false), thunder time: 0 (now: false)
	Level game mode: Game mode: creative (ID 1). Hardcore: false. Cheats: false
	Forced entities: 28 total; [EntityChicken['雞'/0, l='MpServer', x=-624.21, y=4.00, z=276.13], EntityCow['牛'/5, l='MpServer', x=-588.18, y=4.00, z=274.38], EntityCow['牛'/6, l='MpServer', x=-588.23, y=4.00, z=283.17], EntitySheep['綿羊'/7, l='MpServer', x=-579.67, y=4.00, z=272.20], EntityCow['牛'/8, l='MpServer', x=-568.56, y=4.00, z=296.17], EntityCow['牛'/18, l='MpServer', x=-538.81, y=4.00, z=279.79], EntityCow['牛'/19, l='MpServer', x=-538.82, y=4.00, z=284.35], EntityWorm['Worm'/23, l='MpServer', x=-560.50, y=3.50, z=244.50], EntityWorm['Worm'/24, l='MpServer', x=-561.50, y=3.50, z=246.50], EntityItem['item.tile.dirt.default'/25, l='MpServer', x=-558.13, y=3.00, z=246.37], EntityItem['item.item.actuallyadditions.item_paxel_crystal_green'/26, l='MpServer', x=-559.16, y=3.94, z=243.77], EntityItem['item.tile.dirt.default'/27, l='MpServer', x=-559.27, y=4.00, z=242.66], EntityWorm['Worm'/28, l='MpServer', x=-559.50, y=3.50, z=247.50], EntityItem['item.item.actuallyadditions.item_worm'/29, l='MpServer', x=-558.47, y=3.00, z=248.16], EntityItem['item.item.actuallyadditions.item_worm'/30, l='MpServer', x=-559.79, y=3.94, z=248.80], EntityItem['item.item.rftools_shape_manual'/31, l='MpServer', x=-557.46, y=4.00, z=251.18], EntityItem['item.item.rftools_manual'/32, l='MpServer', x=-557.33, y=4.00, z=251.10], EntityPlayerSP['Mu_guang_138'/22, l='MpServer', x=-560.88, y=4.82, z=250.87], EntitySheep['綿羊'/34, l='MpServer', x=-636.50, y=4.00, z=257.50], EntitySheep['綿羊'/35, l='MpServer', x=-636.54, y=4.00, z=259.78], EntityChicken['雞'/36, l='MpServer', x=-624.14, y=4.00, z=266.90], EntityHorse['馬'/37, l='MpServer', x=-637.34, y=4.00, z=261.16], EntitySheep['綿羊'/38, l='MpServer', x=-630.81, y=4.00, z=252.62], EntityCow['牛'/45, l='MpServer', x=-542.31, y=4.00, z=254.82], EntityChicken['雞'/51, l='MpServer', x=-611.84, y=4.00, z=261.68], EntityChicken['雞'/52, l='MpServer', x=-614.21, y=4.00, z=266.48], EntityItem['item.item.egg'/53, l='MpServer', x=-608.43, y=4.00, z=265.87], EntityCow['牛'/54, l='MpServer', x=-594.46, y=4.00, z=251.83]]
	Retry entities: 0 total; []
	Server brand: fml,forge
	Server type: Integrated singleplayer server
Stacktrace:
	at net.minecraft.client.multiplayer.WorldClient.func_72914_a(WorldClient.java:532)
	at net.minecraft.client.Minecraft.func_71396_d(Minecraft.java:2748)
	at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:426)
	at net.minecraft.client.main.Main.main(SourceFile:123)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at net.minecraft.launchwrapper.Launch.launch(Launch.java:135)
	at net.minecraft.launchwrapper.Launch.main(Launch.java:28)

-- System Details --
Details:
	Minecraft Version: 1.12.2
	Operating System: Windows 10 (amd64) version 10.0
	Java Version: 1.8.0_25, Oracle Corporation
	Java VM Version: Java HotSpot(TM) 64-Bit Server VM (mixed mode), Oracle Corporation
	Memory: 1153715312 bytes (1100 MB) / 2147483648 bytes (2048 MB) up to 2147483648 bytes (2048 MB)
	JVM Flags: 8 total; -XX:HeapDumpPath=MojangTricksIntelDriversForPerformance_javaw.exe_minecraft.exe.heapdump -Xmx2G -XX:+UnlockExperimentalVMOptions -XX:+UseG1GC -XX:G1NewSizePercent=20 -XX:G1ReservePercent=20 -XX:MaxGCPauseMillis=50 -XX:G1HeapRegionSize=10G
	IntCache: cache: 0, tcache: 0, allocated: 0, tallocated: 0
	FML: MCP 9.42 Powered by Forge 14.23.0.2528 Optifine OptiFine_1.12.2_HD_U_C6 24 mods loaded, 24 mods active
	States: 'U' = Unloaded 'L' = Loaded 'C' = Constructed 'H' = Pre-initialized 'I' = Initialized 'J' = Post-initialized 'A' = Available 'D' = Disabled 'E' = Errored

	| State     | ID                  | Version                  | Source                                      | Signature                                |
	|:--------- |:------------------- |:------------------------ |:------------------------------------------- |:---------------------------------------- |
	| UCHIJAAAA | minecraft           | 1.12.2                   | minecraft.jar                               | None                                     |
	| UCHIJAAAA | mcp                 | 9.42                     | minecraft.jar                               | None                                     |
	| UCHIJAAAA | FML                 | 8.0.99.99                | forge-1.12.2-14.23.0.2528.jar               | e3c3d50c7c986df74c645c0ac54639741c90a557 |
	| UCHIJAAAA | forge               | 14.23.0.2528             | forge-1.12.2-14.23.0.2528.jar               | e3c3d50c7c986df74c645c0ac54639741c90a557 |
	| UCHIJAAAA | mercurius_updater   | 1.0                      | MercuriusUpdater-1.12.2.jar                 | None                                     |
	| UCHIJAAAA | actuallyadditions   | 1.12.2-r123              | ActuallyAdditions-1.12.2-r123.jar           | None                                     |
	| UCHIJAAAA | astralsorcery       | 1.7.2                    | astralsorcery-1.12.2-1.7.2.jar              | a0f0b759d895c15ceb3e3bcb5f3c2db7c582edf0 |
	| UCHIJAAAA | psi                 | r1.1-58                  | Psi-r1.1-58.jar                             | None                                     |
	| UCHIJAAAA | autoreglib          | 1.3-14                   | AutoRegLib-1.3-14.jar                       | None                                     |
	| UCHIJAAAA | baubles             | 1.5.1                    | Baubles-1.12-1.5.1.jar                      | None                                     |
	| UCHIJAAAA | botania             | r1.10-349                | Botania r1.10-349.jar                       | None                                     |
	| UCHIJAAAA | codechickenlib      | 3.1.3.313                | CodeChickenLib-1.12-3.1.3.313-universal.jar | f1850c39b2516232a2108a7bd84d1cb5df93b261 |
	| UCHIJAAAA | chickenchunks       | 2.4.0.70                 | ChickenChunks-1.12-2.4.0.70-universal.jar   | f1850c39b2516232a2108a7bd84d1cb5df93b261 |
	| UCHIJAAAA | enderstorage        | 2.4.0.122                | EnderStorage-1.12-2.4.0.122-universal.jar   | f1850c39b2516232a2108a7bd84d1cb5df93b261 |
	| UCHIJAAAA | fpsreducer          | 1.12.2-1.7               | FpsReducer-1.12.2-1.7.jar                   | None                                     |
	| UCHIJAAAA | guideapi            | 1.12-2.1.4-57            | Guide-API-1.12-2.1.4-57.jar                 | None                                     |
	| UCHIJAAAA | waila               | 1.8.21                   | Hwyla-1.8.21-B36_1.12.jar                   | None                                     |
	| UCHIJAAAA | inventorytweaks     | 1.63+release.109.220f184 | InventoryTweaks-1.63.jar                    | 55d2cd4f5f0961410bf7b91ef6c6bf00a766dcbe |
	| UCHIJAAAA | jei                 | 4.8.0.105                | jei_1.12.2-4.8.0.105.jar                    | None                                     |
	| UCHIJAAAA | nei                 | ${mod_version}           | NotEnoughItems-1.12-2.4.0.231-universal.jar | f1850c39b2516232a2108a7bd84d1cb5df93b261 |
	| UCHIJAAAA | projecte            | 1.12-PE1.1.0             | ProjectE-1.12-PE1.1.0.jar                   | None                                     |
	| UCHIJAAAA | refinedstorage      | 1.5.17                   | refinedstorage-1.5.17.jar                   | None                                     |
	| UCHIJAAAA | rftools             | 7.12                     | rftools-1.12-7.12.jar                       | None                                     |
	| UCHIJAAAA | wailaharvestability | 1.1.12                   | WailaHarvestability-mc1.12-1.1.12.jar       | None                                     |

	Loaded coremods (and transformers): 
Inventory Tweaks Coremod (InventoryTweaks-1.63.jar)
  invtweaks.forge.asm.ContainerTransformer
AstralCore (astralsorcery-1.12.2-1.7.2.jar)
  
	GL info: ' Vendor: 'Intel' Version: '4.5.0 - Build 22.20.16.4749' Renderer: 'Intel(R) HD Graphics 630'
	Launched Version: 1.12.2-forge1.12.2-14.23.0.2528
	LWJGL: 2.9.4
	OpenGL: Intel(R) HD Graphics 630 GL version 4.5.0 - Build 22.20.16.4749, Intel
	GL Caps: Using GL 1.3 multitexturing.
Using GL 1.3 texture combiners.
Using framebuffer objects because OpenGL 3.0 is supported and separate blending is supported.
Shaders are available because OpenGL 2.1 is supported.
VBOs are available because OpenGL 1.5 is supported.

	Using VBOs: Yes
	Is Modded: Definitely; Client brand changed to 'fml,forge'
	Type: Client (map_client.txt)
	Resource Packs: 
	Current Language: 繁體中文 (台灣)
	Profiler Position: N/A (disabled)
	CPU: 4x Intel(R) Core(TM) i5-7500 CPU @ 3.40GHz
	OptiFine Version: OptiFine_1.12.2_HD_U_C6
	Render Distance Chunks: 16
	Mipmaps: 0
	Anisotropic Filtering: 1
	Antialiasing: 0
	Multitexture: false
	Shaders: null
	OpenGlVersion: 4.5.0 - Build 22.20.16.4749
	OpenGlRenderer: Intel(R) HD Graphics 630
	OpenGlVendor: Intel
	CpuCount: 4

**Comment by Quarris** [comment]
Im sorry for this. I can take the blame.
https://gist.github.com/Quarris/d75482eb2c31f3603835455d9cfd874a
